Sources/Kitura/RouterResponse.swift

```diff
@@ -178,6 +178,116 @@ public class RouterResponse {
             }
         }
     }
+    
+    /// An enum to describe the attributes  (name, value, domain, path etc) of a cookie.
+    public enum CookieAttribute {
```

Comment:
It looks like HTTPCookiePropertyKey allows you to define custom cookie property keys, can we add support for that behavior?

Is it possible that the available HTTPCookiePropertyKey value set changes in the future? If so then maybe an alternative to enum would be a good idea so we avoid the need for a major version due to changing a public enum.

Reply:
Thanks for the review .I will find a way to add support for defining custom cookie property keys. I also work on keeping the enum extensible.

Reply:
Thanks. Given the API exists it is probably better to expose it, particularly as we are sticking with an enum!

Reply:
I'm not convinced that we should be introducing a public enum as a proxy for something that is not under our control (even though Foundation is unlikely to change), it leaves us exposed if we were forced to change it for some reason.

You could instead create a Struct with the same semantics as the enum, using an internal enum (so we can change it in the future without breaking our public API).  It looks something like this:

```swift
public struct CookieAttribute {
    internal enum _CookieAttribute {
        case portList([NSNumber]?)
        ... etc
    }
    internal let _value: CookieAttribute

    private init(_ value: _CookieAttribute) {
        self._value = value
    }

    public static func portList(_ value: [NSNumber]?) -> CookieAttribute {
        return CookieAttribute(_CookieAttribute(value))
    }
    ...etc
}
```